lib/diagnostic.js

```diff
@@ -6,33 +6,78 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+
+const getNormalizedWords = function(string) {
+  // split the string into an array of words
+  let words = string.split(/\s+/);
+  // normalize these words
+  let normalizedWords = [];
+  for (let i = 0; i < words.length; i++) {
+    let word = words[i].replace(/\W+/, '');
+    normalizedWords[i] = word.toUpperCase();
+  }
+  return normalizedWords;
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const getUniqueWords = function(string) {
+  let normalizedWords = getNormalizedWords(string);
+//create object of unique words
+  let uniqueWordsAsKeys = {};
+// The words stored as keys of a dictionary
+    for (let i = 0; i < normalizedWords.length; i++) {
+      let word = normalizedWords[i];
+      uniqueWordsAsKeys[word] = true;
+  }
+// The keys pulled out into a list
+let uniqueWords = [];
+
+let keys = [];
+let k = 0;
+
+for(keys[k++] in uniqueWordsAsKeys); // jshint ignore:line
+
+uniqueWords = keys;
 };
```

Comment:
You are totally on the right track with your strategy. In line 31, you are storing the unique words in a dictionary with `true` as the value. But actually, what you want to accomplish is to store a number that increments every time that word occurs. That way you wouldn't need the 2nd for loop. Does that make sense?

FYI, the best solution for this problem would have been to piggy-back off of the `wordFrequencies` function.